data/campaigns/Under_the_Burning_Suns/utils/abilities.cfg

```diff
@@ -1,5 +1,78 @@
 #textdomain wesnoth-help
 
+#define STR_USES_ATTACKS_NAME_1
+_ "uses 1 attack"#enddef
+
+#define STR_USES_ATTACKS_NAME_2
+_ "uses 2 attacks"#enddef
+
+#define STR_USES_ATTACKS_NAME_3
+_ "uses 3 attacks"#enddef
+
+#define STR_USES_ATTACKS_NAME_4
+_ "uses 4 attacks"#enddef
+
+#define STR_USES_ATTACKS_NAME_5
+_ "uses 5 attacks"#enddef
+
+#define STR_USES_ATTACKS_NAME_6
+_ "uses 6 attacks"#enddef
+
+#define STR_USES_ATTACKS_RESTRICT_1
+_ "This attack uses 1 attack"#enddef
+
+#define STR_USES_ATTACKS_RESTRICT_2
+_ "This attack uses 2 attacks"#enddef
+
+#define STR_USES_ATTACKS_RESTRICT_3
+_ "This attack uses 3 attacks"#enddef
+
+#define STR_USES_ATTACKS_RESTRICT_4
+_ "This attack uses 4 attacks"#enddef
+
+#define STR_USES_ATTACKS_RESTRICT_5
+_ "This attack uses 5 attacks"#enddef
+
+#define STR_USES_ATTACKS_RESTRICT_6
+_ "This attack uses 6 attacks"#enddef
+
+#define WEAPON_SPECIAL_USES_ATTACKS X
+    [disable]
+        id = uses_attacks
+        name = {STR_USES_ATTACKS_NAME_{X}}
+        description = {STR_USES_ATTACKS_RESTRICT_{X}}
+        [filter_self]
+            formula = "self.max_attacks < {X}"
```

Comment:
Shouldn't this be `self.attacks_left`?

Reply:
Oh I cant filter it by ``self.attacks_left`` because then attack will stop before it even happens because attacks get consumed and attack is instantly disabled because condition for disabling is fullfilled thus using ``self.max_attacks`` is actually a working option.

Reply:
Your explanation is pretty incoherent, but let me see if I've understood it correctly.

If you use `attacks_left`, the game will allow you to use the attack, but then the attack will fail because the state of the filter changes between the time you choose the attack and the time the attack is executed.

Is that correct?

Reply:
I guess yes.
Like you try to attack -> attacks are consumed -> disable turns on because you no longer meet the required amount of attacks -> attack is disabled.

Reply:
Hmm… that seems really unfortunate… I'll see if I can think of a better solution but I guess this works for now.

Reply:
Would this `[disable]` and the related strings be needed at all if the engine's handling of `attacks_left` was fixed?

If it needs to be fixed in the engine, it would be better to have the campaign testing the engine code all the time instead of adding a workaround in the campaign.

Reply:
I think something should be done in the engine but I haven't worked out what that would look like. The most obvious option (deducting from `moves` and `attacks_left` only after the attack completes) is probably a significant compatibility issue and can't easily be considered. Other options I've considered are adding a formula variable `is_attacking` which is only true literally in the middle of an attack (so the filter becomes `is_attacking or attacks_left < attacks_used`) or adding a new key in the weapon to toggle between the two behaviours (something like `needs_exact_attacks=true` to select the UtBS-required behaviour).

Regarding the strings, those are presumably a way to deal with #7367, so I assume they wouldn't be needed if that was addressed.

Reply:
Do we need it to affect anything other than the UI when the user is selecting an attack, and the list when the AI is selecting an attack?

Adding a new key/attribute and then just affecting those two points sounds simpler (without looking into the details) than anything which needs to handle special cases once the attack has already been chosen.

Reply:
I'm not quite sure what you're trying to say but it feels like you've missed a point somewhere in here.

The UI for selecting an attack is fine. The filter states that the attack is enabled, so it lets you pick it.

Then the attack system deducts `attacks_used` from the current `attacks_left`.

Now the filter states that the attack is disabled, so the attack system aborts the attack and nothing happens.

Reply:
When is this `[disable]` needed? Could the entire `[disable]` be removed?

If the UI only lets the user pick attacks that have enough `attack_left`, then why does the attack need a `[disable]` at all?

Reply:
That does solve this specific use-case. I do think that using `[disable]` (or any ability) based on `attacks_left` or `moves` should ideally function rather than causing an abort of the attack, but that wouldn't be needed in this specific case if there were a key that specifies the attack cannot be executed if there are insufficient `attacks_left`. That's the third possible solution I noted in my earlier post (`needs_exact_attacks=true`).

Reply:
Once the attack has been selected (and possibly after any `pre_attack` event has fired), I think any logic based on `attacks_left` or `moves` should see the value after the cost of attacking has been deducted. That would simplify scripting dialogue such as the unit saying "I can attack again", or even animation where the unit is breathing more heavily when they've already attacked.

However, for this PR my main point is that I'd prefer the mainline campaigns to use the engine's features rather than coding workarounds for them. Otherwise we'll release 1.18 and then find that part of the engine wasn't tested and all UMC ends up copying the workaround too.

Reply:
I think that makes sense, yeah, but in this one case it doesn't make sense, and I imagine you could come up with other cases where it doesn't make sense. Perhaps the second solution's better? Then the formula could be forced to be true while the attack is in progress, and while it's not, it's correctly based on `moves` or `attacks_left`.

Reply:
> I think that makes sense, yeah, but in this one case it doesn't make sense, and I imagine you could come up with other cases where it doesn't make sense. Perhaps the second solution's better? Then the formula could be forced to be true while the attack is in progress, and while it's not, it's correctly based on `moves` or `attacks_left`.

Another option is to not evaluate the filter at all in the middle of the attack process. I don't know if that's viable though. Maybe some things rely on the possibility of it changing to abort the attack.